src/app/frontend/deploy/deployfromsettings_controller.js

```diff
@@ -183,6 +190,33 @@ export default class DeployFromSettingsController {
   }
 
   /**
+   * @returns {!angular.$q.Promise}
+   * @export
+   */
+  updateAvailableProtocols() {
```

Comment:
Yeah, I'm sorry I didn't make this comment at first. It only arose after I've been testing this on my machine. 

The validator solution seems better and better to me the more I think about it. Since this is like saying: "This protocol is invalid for external services", which looks good.